src/librustc_resolve/lib.rs

```diff
@@ -4796,10 +4796,18 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
                 err.span_suggestion_with_applicability(
                     binding.span,
                     rename_msg,
-                    if snippet.ends_with(';') {
-                        format!("{} as {};", &snippet[..snippet.len() - 1], suggested_name)
+                    if snippet.contains(" as ") {
+                        format!(
+                            "{} as {}",
+                            &snippet[..snippet.find(" as ").unwrap()],
+                            suggested_name,
+                        )
                     } else {
-                        format!("{} as {}", snippet, suggested_name)
+                        if snippet.ends_with(';') {
+                            format!("{} as {};", &snippet[..snippet.len() - 1], suggested_name)
+                        } else {
+                            format!("{} as {}", snippet, suggested_name)
+                        }
                     },
                     Applicability::MachineApplicable,
```

Comment:
I don't think this should be machine applicable. The change is harmless, but the name provided is meaningless, let's let the developer decide what is an appropriate rename. Change to `MaybeIncorrect`.